src/content/docs/extensions/algo/kcore.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: K-Core Decomposition
+---
+
+The k-core decomposition identifies subgraphs where every node has a degree of at least `k` within the subgraph. 
+In other words, a k-core is a maximal subgraph where each node is connected to at least `k` other nodes in the same subgraph. 
+The k-core decomposition process involves iteratively removing nodes with degree less than `k` until no more
+nodes can be removed, with the remaining nodes forming the k-core. 
+
+K-Core decomposition is useful in domains like social network analysis, where it can help identify
+cohesive groups, and in network resilience analysis, where it can help understand the robustness of a network to node removal.
+
+Kuzu implements a parallelized version of K-Core decomposition based on [Ligra](https://jshun.csail.mit.edu/ligra.shtml).
+
+## Syntax
+
+```
+CALL k_core_decomposition(
```

Comment:
I now feel we'll be making users type out these long function names unnecessarily.
Something like `kcore` or maybe `algo_kcore`, `algo_wcc`, `algo_scc`, `algo_pr`, `algo_louvain` could have been better.

Reply:
Yeah. same feeling.
I think `louvain` and `page_rank` are fine. But we should provide alias for `k_core_decomposition`, `weakly_connected_components`, `strongly_connected_components`, and `strongly_connected_components_kosaraju`.
Maybe simply `kcore`, `wcc`, `scc`, and `kosaraju` are good enough?

Reply:
Added